File: java/Fence363B.java
package com.nicholasmoreles.cpsolutions;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

/**
 * 
 * @author Nicholas Moreles
 */
public class Fence363B {
	
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		final int numPlanks = Integer.parseInt(st.nextToken());
		final int pianoWidth = Integer.parseInt(st.nextToken());
		final int[] planks = new int[numPlanks + 1]; // index 0 is empty
		st = new StringTokenizer(br.readLine());
		
		for (int i = 1; i <= numPlanks; ++i) {
			planks[i] =  Integer.parseInt(st.nextToken());
		}
		
		long lastWidthPlanks = 0;
		int bestIndex = -1;
		long bestPlanks = Long.MAX_VALUE;
		for (int i = 1; i <= numPlanks; ++i) {
			lastWidthPlanks += i > pianoWidth ? planks[i] - planks[i - pianoWidth] : planks[i];
			if (i >= pianoWidth && lastWidthPlanks < bestPlanks) {
				bestPlanks = lastWidthPlanks;
				bestIndex = i - (pianoWidth - 1);
			}
		}
		System.out.println(bestIndex);
	}
}

File: c++/Fence363B.cc
#include <iostream>
#include <limits>
#include <vector>

using std::cin;
using std::cout;
using std::numeric_limits;
using std::vector;

int main() {
  int num_planks, piano_width;
  cin >> num_planks >> piano_width;

  vector<int> planks;
  planks.push_back(0); // dummy value
  for (int i = 1; i <= num_planks; ++i) {
    int temp;
    cin >> temp;
    planks.push_back(temp);
  }

  long last_width_planks = 0;
  long best_planks = numeric_limits<long>::max();
  int best_index = -1;
  for (int i = 1; i <= num_planks; ++i) {
    last_width_planks += (i > piano_width) ? planks[i] - planks[i - piano_width] : planks[i];
    if (i >= piano_width && last_width_planks < best_planks) {
      best_planks = last_width_planks;
      best_index = i - (piano_width - 1);
    }
  }
  cout << best_index;

  return 0;
}
